ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -5266,7 +5266,8 @@ static private boolean needToCopy(final HiveConf conf, Path srcf, Path destf, Fi
     try {
       return srcHdfsEncryptionShim != null
           && destHdfsEncryptionShim != null
-          && (srcHdfsEncryptionShim.isPathEncrypted(srcf) || destHdfsEncryptionShim.isPathEncrypted(destf))
+          && ((srcFs.exists(srcf) && srcHdfsEncryptionShim.isPathEncrypted(srcf))
```

Comment:
why do you need extra brackets for `&&` operands?

Reply:
Thank you for the review @deniskuzZ, updated.